core/src/main/scala/variant/Antichess.scala

```diff
@@ -103,3 +98,9 @@ case object Antichess
     // The pawn cannot attack a bishop or be attacked by a bishop
     val cannotAttackBishop = pawn.isLight != oppositeBishopLight
     InsufficientMatingMaterial.pawnBlockedByPawn(pawn, position) && cannotAttackBishop
+
+  private def AllOnSameColourSquares(position: Position): Boolean =
+    position.white.squares.forall(_.isLight) == position.black.squares.forall(_.isLight)
```

Comment:
the name of this function is misleading.

In a position where both players have pieces on light and dark squares - like the initial position - then `AllOnSameColourSquares` returns true.